### PR TITLE
Support all attribute types with constraints.

### PR DESCRIPTION
**Pull request by brndnmtthws**
If attributes are some non-text type (such as scalar or range),
constraints will not work because we assume all attributes are text.

To address:
 * https://github.com/mesosphere/marathon/issues/3857
 * https://github.com/mesosphere/marathon/issues/1711

### CODE REVIEW

**Comment by air** [comment]
hi @aquamatthias, just a heads up we need this in the next available Marathon release. Thanks.

File: src/main/scala/mesosphere/mesos/Constraints.scala [hunk 1]
```diff
@@ -25,6 +25,21 @@ object Constraints {
...
+      java.text.NumberFormat.getInstance.format(attribute.getScalar.getValue)
+    case Value.Type.TEXT =>
+      attribute.getText.getValue
+    case Value.Type.RANGES =>
+      val s = attribute.getRanges.getRangeList.asScala
+        .map(r => s"${r.getBegin.toString}-${r.getEnd.toString}")
+        .mkString(",")
+      s"[$s]"
+    case Value.Type.SET =>
+      val s = attribute.getSet.getItemList.asScala.mkString(",")
```
Comment (aquamatthias) [inline src/main/scala/mesosphere/mesos/Constraints.scala]: I think we should sort the set. `"a,b" != "b,a"`
With sorting the use at least know how to specify this.
This should also be part of the documentation.

File: src/main/scala/mesosphere/mesos/Constraints.scala [hunk 2]
```diff
@@ -25,6 +25,21 @@ object Constraints {
...
   }
 
+  private def getValueString(attribute: Attribute): String = attribute.getType match {
+    case Value.Type.SCALAR =>
+      java.text.NumberFormat.getInstance.format(attribute.getScalar.getValue)
+    case Value.Type.TEXT =>
+      attribute.getText.getValue
+    case Value.Type.RANGES =>
+      val s = attribute.getRanges.getRangeList.asScala
+        .map(r => s"${r.getBegin.toString}-${r.getEnd.toString}")
```
Comment (aquamatthias) [inline src/main/scala/mesosphere/mesos/Constraints.scala]: I think we should sort the ranges. `"1-2,2-3" != "2-3,1-2"`
With sorting the use at least know how to specify this.
This should also be part of the documentation.

**Comment by aquamatthias** [comment]
@brndnmtthws Thanks for this PR. Added 2 small suggestions.

**Comment by aquamatthias** [comment]
@brndnmtthws thanks

**Comment by bobrik** [comment]
This is great, @brndnmtthws. Can't wait to try the next release.

**Pull request merged**